### PR TITLE
multi: error if restore seed/pubkey already exists for decred wallet

### PR DESCRIPTION
**Pull request by ukane-philemon**
Issue Number (if Applicable): Fixes #

# Description

Check and error if user attempts to create wallet with existing seed/xpub key.

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

**Comment by ukane-philemon** [comment]
TODO: Update `build_decred.sh` scripts for `android`, `ios`, and `macos` when https://github.com/decred/libwallet/pull/8 has been merged and taged.

File: cw_decred/lib/wallet_service.dart [hunk 1]
```diff
@@ -41,6 +41,18 @@ class DecredWalletService extends WalletService<
...
+  Future<bool> checkIfWalletWithSeedExists(String walletName, String walletPassword,
+      DecredRestoreWalletFromSeedCredentials credentials) async {
+    final w = await openWallet(walletName, walletPassword);
+    return credentials.mnemonic == w.seed;
+  }
+
+  Future<bool> checkIfWalletWithKeyExists(String walletName, String walletPassword,
+      DecredRestoreWalletFromPubkeyCredentials credentials) async {
+    final w = await openWallet(walletName, walletPassword);
+    return credentials.pubkey == w.pubkey;
```
Comment (JoeGruffins) [inline cw_decred/lib/wallet_service.dart]: Should we close the wallet afterwards?

**Review by JoeGruffins — COMMENTED** [review]
If I cause it to error then try again it seems to multiply the amount of times I have to confirm on this screen:
image

**Comment by ukane-philemon** [comment]
> If I cause it to error then try again it seems to multiply the amount of times I have to confirm on this screen: image

I've noticed something wonky with that confirmation screen(on mac). If you click okay, it takes the user back to the previous screen but remains on the screen until user clicks outside the modal.

Let me know if you want me to look into it. The correct behaviour is to hide the modal once the use clicks ok.

File: cw_decred/lib/wallet_service.dart (hunk 1 above)
Comment (ukane-philemon) [reply cw_decred/lib/wallet_service.dart]: We risk closing a currently used or previously loaded wallet. IMO, it doesn't hurt to have all decred wallets loaded(in libwallet).
By the time users need them in cake, it wouldn't be an issue.

Comment (ukane-philemon) [reply cw_decred/lib/wallet_service.dart]: Or, better still, we could extend the implementation method to libwallet, which can be reused in other projects. That way we can close the wallets we know were not previously loaded in libwallet.

Comment (JoeGruffins) [reply cw_decred/lib/wallet_service.dart]: I'm not really a fan of the map in the first place. I would like to close the wallet yeah. Also don't want to make any huge changes to libwallet when it seems to be working well atm. Let me think...

Comment (JoeGruffins) [reply cw_decred/lib/wallet_service.dart]: Looking around, it doesn't look like they ever close wallets. So maybe it's fine...

Could you leave a message "TODO: Consider closing wallets."? And we will not worry about it for now.

**Comment by JoeGruffins** [comment]
> Let me know if you want me to look into it. The correct behaviour is to hide the modal once the use clicks ok.

If it's their bug then don't worry about it.

Can you just add the close wallet TODO message for me? Then I think this is good. Still thinking about the libwallet changes but I think this pr is fine without them.

**Comment by JoeGruffins** [comment]
Thanks! Added to the main pr and testing pr.

**Pull request closed without merging**